hooks/provider-pinned-versions/required_providers.awk

```diff
@@ -0,0 +1,54 @@
+#! /bin/awk
```

Comment:
AWK is fine, but I wonder if it would be easier if we used python (since it's available already as pre-commit uses python)?

Reply:
I thought of it as well but I thought we wouldn't have access to it since we're removing the `.python-version` files.

Reply:
We can leave that in... and/or just assume that python is present.